src/Antd_DatePicker.re

```diff
@@ -0,0 +1,434 @@
+      ~showToday: Js.boolean=?,
+      ~_open: Js.boolean=?,
+      ~disabledTime: moment => 'b=?, /* todo: (current: moment.Moment) => { disabledHours?: () => number[],disabledMinutes?: () => number[], disabledSeconds?: () => number[],}; */
+      ~onOpenChange: Js.boolean => unit=?,
+      ~onOk: moment => unit=?,
+      ~placeholder: string=?,
+      /*export interface SinglePickerProps*/
+      ~value: moment=?,
+      ~defaultValue: moment=?,
+      ~defaultPickerValue: moment=?,
+      ~onChange: (moment, string) => unit=?,
+      /*export interface PickerProps*/
+      ~prefixCls: string=?,
+      ~inputPrefixCls: string=?,
+      ~format: string=?,
+      ~disabled: Js.boolean=?,
+      ~allowClear: Js.boolean=?,
+      ~popupStyle: ReactDOMRe.Style.t=?,
+      ~locale: 'c=?,
+      ~size: string=?,
+      ~renderExtraFooter: unit => ReasonReact.reactElement=?,
+      ~getCalendarContainer: Dom.element => Dom.htmlElement=?,
+      ~disabledDate: moment => bool=?,
+      ~dateRender: (moment, moment) => ReasonReact.reactElement=?,
+      ~id: string=?,
+      ~className: string=?,
+      ~style: ReactDOMRe.Style.t=?,
+      unit
+    ) =>
+    t(_) =
+    "";
+};
+
+let make =
+    (
+      ~showTime=?,
+      ~showToday=?,
+      ~_open=?,
+      ~disabledTime=?,
+      ~onOpenChange=?,
+      ~onOk=?,
+      ~placeholder=?,
+      ~value=?,
+      ~defaultValue=?,
+      ~defaultPickerValue=?,
+      ~onChange=?,
+      ~prefixCls=?,
+      ~inputPrefixCls=?,
+      ~format=?,
+      ~disabled=?,
+      ~renderExtraFooter=?,
+      ~allowClear=?,
+      ~popupStyle=?,
+      ~locale=?,
+      ~size=?,
+      ~getCalendarContainer=?,
+      ~disabledDate=?,
+      ~dateRender=?,
+      ~id=?,
+      ~className=?,
+      ~style=?,
+      children
+    ) =>
+  ReasonReact.wrapJsForReason(
+    ~reactClass,
+    ~props=
+      DatePickerProps.make(
+        ~showTime=?Js.Option.map([@bs] (b => boolOrPropsToProp(b)), showTime),
+        ~showToday=?Js.Option.map([@bs] (b => from_bool(b)), showToday),
+        ~_open=?Js.Option.map([@bs] (b => from_bool(b)), _open),
+        ~disabledTime?,
+        ~onOpenChange?,
+        ~onOk?,
+        ~placeholder?,
+        ~value?,
+        ~defaultValue?,
+        ~defaultPickerValue?,
+        ~onChange?,
+        ~prefixCls?,
+        ~inputPrefixCls?,
+        ~format?,
+        ~disabled=?Js.Option.map([@bs] (b => from_bool(b)), disabled),
+        ~renderExtraFooter?,
+        ~allowClear=?Js.Option.map([@bs] (b => from_bool(b)), allowClear),
+        ~popupStyle?,
+        ~locale?,
+        ~size=?Js.Option.map([@bs] (b => pickerSizeToJs(b)), size),
+        ~getCalendarContainer?,
+        ~disabledDate?,
+        ~dateRender?,
+        ~id?,
+        ~className?,
+        ~style?,
+        ()
+      ),
+    children
+  );
+
+module RangePicker = {
+  [@bs.module "antd/lib/date-picker"]
+  external reactClass : ReasonReact.reactClass = "RangePicker";
+  module RangePickerProps = {
```

Comment:
I’m not sure why do you create a new module here? Type t is understandable for root RangerPicker IMHO

Reply:
consistency with TimePickerProps I guess,  only difference is two lines and some indentation.

the resultant js is also pretty much the same.

Reply:
I would remove the extra module in both, because that would be annoying to type it use the make function. And please use makeProps for all modules. It’s a convention to make sure you always know what to type